src/libutil/util.hh

```diff
@@ -368,22 +369,33 @@ MakeError(FormatError, Error);
 
 
 /* String tokenizer. */
-template<class C> C tokenizeString(std::string_view s, const string & separators = " \t\n\r");
+template<class C> C tokenizeString(string_view s, string_view separators = " \t\n\r");
 
 
 /* Concatenate the given strings with a separator between the
    elements. */
 template<class C>
-string concatStringsSep(const string & sep, const C & ss)
+string concatStringsSep(const string_view sep, const C & ss)
 {
     string s;
+    s.reserve(
+        std::accumulate(std::begin(ss), std::end(ss), string::size_type(0),
+            [] (auto size, string_view s) { return size + s.size(); }));
```

Comment:
Call me old-fashioned but I don't really like these STL algorithms like `std::accumulate` since they're verbose and non-functional (e.g. you have to mention `ss` multiple times, so you can't use it on an arbitrary expression). I prefer just
```
size_t size = 0;
for (auto & s : ss) size += s.size();
```

Reply:
yeah, they are the opposite of pretty. at least C++20 has ranges, but even then `accumulate` is still pretty ugly :/